prow/github/client.go

```diff
@@ -1568,6 +1569,23 @@ func (c *client) GetBranches(org, repo string, onlyProtected bool) ([]Branch, er
 	return branches, nil
 }
 
+// GetBranchProtection returns current protection object for the branch
+//
+// See https://developer.github.com/v3/repos/branches/#get-branch-protection
+func (c *client) GetBranchProtection(org, repo, branch string) (*BranchProtection, error) {
+	c.log("GetBranchProtection", org, repo, branch)
+	var retProtection BranchProtection
+	_, err := c.request(&request{
+		method:    http.MethodGet,
+		path:      fmt.Sprintf("/repos/%s/%s/branches/%s/protection", org, repo, branch),
```

Comment:
This endpoint appears to return 404s if the branch has no protection configured which will cause the function to incorrectly return an error. Unfortunately the endpoint also appears to return a 404 if the access token used by the request does not have sufficient privileges so this function will also need to check if the message contains `Branch not protected` to distinguish between legitimate errors and no protection configured.

Reply:
Thanks, I will update the function to differentiate 404 due to `Branch not protected` and 404 due to any other reason.